src/Examples/Flowsheets/power_generation/rsofc/cpu.py

```diff
@@ -0,0 +1,720 @@
+            )
+
+        # Water stream: O2 mole flow
+        @self.Constraint(
+            self.flowsheet().config.time, doc="o2 component flow_mol in water"
+        )
+        def water_o2_flow_mol_comp_eq(b, t):
+            return b.water_flow_mol_comp[t, "O2"] == sm.water_o2_flow_mol_comp_fun.f(
+                b.inlet_flow_mol[t],
+                b.inlet_mole_frac_comp[t, "Ar"],
+                b.inlet_mole_frac_comp[t, "CO2"],
+                b.inlet_mole_frac_comp[t, "O2"],
+                b.inlet_mole_frac_comp[t, "H2O"],
+                b.inlet_mole_frac_comp[t, "N2"],
+            )
+
+        # Water stream: AR mole flow
+        @self.Constraint(
+            self.flowsheet().config.time, doc="ar component flow_mol in water"
+        )
+        def water_ar_flow_mol_comp_eq(b, t):
+            return b.water_flow_mol_comp[t, "Ar"] == sm.water_ar_flow_mol_comp_fun.f(
+                b.inlet_flow_mol[t],
+                b.inlet_mole_frac_comp[t, "Ar"],
+                b.inlet_mole_frac_comp[t, "CO2"],
+                b.inlet_mole_frac_comp[t, "O2"],
+                b.inlet_mole_frac_comp[t, "H2O"],
+                b.inlet_mole_frac_comp[t, "N2"],
+            )
+
+        # Water stream: H2O mole flow
+        @self.Constraint(
+            self.flowsheet().config.time, doc="h2o component flow_mol in water"
+        )
+        def water_h2o_flow_mol_comp_eq(b, t):
+            return b.water_flow_mol_comp[t, "H2O"] == sm.water_h2o_flow_mol_comp_fun.f(
+                b.inlet_flow_mol[t],
+                b.inlet_mole_frac_comp[t, "Ar"],
+                b.inlet_mole_frac_comp[t, "CO2"],
+                b.inlet_mole_frac_comp[t, "O2"],
+                b.inlet_mole_frac_comp[t, "H2O"],
+                b.inlet_mole_frac_comp[t, "N2"],
+            )
+
+        # Vent stream: Temperature
+        @self.Constraint(self.flowsheet().config.time, doc="Temperature: vent [K]")
+        def vent_temperature_eq(b, t):
+            return b.vent_temperature[t] == sm.vent_temperature_fun.f(
+                b.inlet_flow_mol[t],
+                b.inlet_mole_frac_comp[t, "Ar"],
+                b.inlet_mole_frac_comp[t, "CO2"],
+                b.inlet_mole_frac_comp[t, "O2"],
+                b.inlet_mole_frac_comp[t, "H2O"],
+                b.inlet_mole_frac_comp[t, "N2"],
+            )
+
+        # Vent stream: Pressure
+        @self.Constraint(self.flowsheet().config.time, doc="Pressure: vent [Pa]")
+        def vent_pressure_eq(b, t):
+            return b.vent_pressure[t] == sm.vent_pressure_fun.f(
+                b.inlet_flow_mol[t],
+                b.inlet_mole_frac_comp[t, "Ar"],
+                b.inlet_mole_frac_comp[t, "CO2"],
+                b.inlet_mole_frac_comp[t, "O2"],
+                b.inlet_mole_frac_comp[t, "H2O"],
+                b.inlet_mole_frac_comp[t, "N2"],
+            )
+
+    def initialize(blk, outlvl=idaeslog.NOTSET, solver="ipopt", optarg={"tol": 1e-6}):
```

Comment:
1. `solver=None` would be preferred, along with using the `get_solver` function below.
2. `optarg=None` would also be preferred, otherwise you need to be a little bit careful to prevent options from seeping into other solves unintentionally (mainly is using the `get_solver` function).